queue/queue_service.js

```diff
@@ -1,7 +1,78 @@
 queueApp.factory("QueueService", function($http){
 	return {
 		getAppointments: function(date){
-			return $http.get("/server/queue.php?displayDate=" + date).then(function(response){
+			let timezoneOffset = new Date().getTimezoneOffset() / 60;
+			return $http.get("/server/queue.php?displayDate=" + date + "&timezoneOffset=" + timezoneOffset).then(function(response){
+				return response.data;
+			},function(error){
+				return null;
+			});
+		},
+		checkInNow: function(time, id) {
+			return $http({
+				url: "/server/queue_priv.php",
+				method: 'POST',
+				data: `action=checkIn&time=${time}&id=${id}`,
+				headers: {
+					'Content-Type': "application/x-www-form-urlencoded"
+				}
+			}).then(function(response){
+				return response.data;
+			},function(error){
+				return null;
+			});
+		},
+		turnInPapers: function(time, id) {
+			return $http({
+				url: "/server/queue_priv.php",
+				method: 'POST',
+				data: `action=completePaperwork&time=${time}&id=${id}`,
+				headers: {
+					'Content-Type': "application/x-www-form-urlencoded"
+				}
+			}).then(function(response){
+				return response.data;
+			},function(error){
+				return null;
+			});
+		},
+		beginAppointment: function(time, id) {
+			return $http({
+				url: "/server/queue_priv.php",
+				method: 'POST',
+				data: `action=appointmentStart&time=${time}&id=${id}`,
+				headers: {
+					'Content-Type': "application/x-www-form-urlencoded"
+				}
+			}).then(function(response){
+				return response.data;
+			},function(error){
+				return null;
+			});
+		},
+		finishAppointment: function(time, id) {
+			return $http({
+				url: "/server/queue_priv.php",
+				method: 'POST',
+				data: `action=appointmentComplete&time=${time}&id=${id}`,
+				headers: {
+					'Content-Type': "application/x-www-form-urlencoded"
+				}
+			}).then(function(response){
+				return response.data;
+			},function(error){
+				return null;
+			});
+		},
+		incompleteAppointment: function(explanation, id) {
+			return $http({
+				url: "/server/queue_priv.php",
+				method: 'POST',
+				data: `action=appointmentIncomplete&explanation=${explanation}&id=${id}`,
```

Comment:
There is a pretty subtle bug here I believe. Imagine that within the explanation, I type the phrase "He was blah blah & blah blah". That & special character is actually going to screw up this URL, and your explanation won't be right in the backend. Also, imagine I did inside the explanation "&id=#", idk how it would handle the situation where there are two id things, but this is something we need to fix. See this post on using the encodeURIComponent: https://stackoverflow.com/a/15287088